protocols/identify/src/identify.rs

```diff
@@ -743,8 +745,21 @@ mod tests {
 
         swarm2.disconnect_peer_id(swarm1_peer_id).unwrap();
 
-        // we should still be able to dial now!
-        swarm2.dial(&swarm1_peer_id).unwrap();
+        // Wait for connection to close.
+        async_std::task::block_on(async {
+            loop {
+                if let SwarmEvent::ConnectionClosed { peer_id, .. } =
+                    swarm2.select_next_some().await
+                {
+                    break peer_id;
+                }
+            }
+        });
+
+        // We should still be able to dial now!
+        swarm2
+            .dial(DialOpts::peer_id(swarm1_peer_id).build())
```

Comment:
`impl From<PeerId> for DialOpts` too?

Reply:
:+1: good idea. Done via https://github.com/libp2p/rust-libp2p/pull/2317/commits/09293c4d9697d1d883dd69d935c481c2c8a2337c.